dqc_us_rules/dqc_us_0011.py

```diff
@@ -0,0 +1,133 @@
+# (c) Copyright 2015 - 2017, XBRL US Inc. All rights reserved.
+# See https://xbrl.us/dqc-license for license information.
+# See https://xbrl.us/dqc-patent for patent infringement notice.
+import os
+import csv
+from .util import messages
+from arelle.PythonUtil import attrdict
+from arelle.ValidateXbrlCalcs import roundFact
+
+
+_CODE_NAME = 'DQC.US.0011'
+_RULE_VERSION = '4.0.0'
+_DQC_11_ITEMS_FILE = os.path.join(
+    os.path.dirname(__file__),
+    'resources',
+    'DQC_US_0011',
+    'dqc_0011.csv'
+)
+_NO_FACT_KEY = 'no_fact'
+
+
+def run_checks(val):
+    """
+    Entrypoint for the rule.  Load the config, search for instances of
+    reversed calculation relationships.
+
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :return: No direct return
+    :rtype: None
+    """
+    model_xbrl = val.modelXbrl
+    for check in _load_checks(model_xbrl):
+        rule_index_key = check.rule_num
+        try:  # allow exceptions when no fact or concept for QName
+            for n_fact in model_xbrl.factsByQname[check.nondim_concept]:
+                # here want dimensionless line items only
+                if not n_fact.context.qnameDims:
+                    # find fact expressed with dimensions
```

Comment:
Perhaps we could do a list comprehension here to remove a layer of indentation. 

n_facts = [fact for fact in model_xbrl.factsByQname[check.nondim_concept] if not n_fact.context.qnameDims]